crates/bevy_reflect/src/tuple.rs

```diff
@@ -496,15 +507,15 @@ macro_rules! impl_reflect_tuple {
 
             #[inline]
             fn clone_dynamic(&self) -> DynamicTuple {
-                let mut dyn_tuple = DynamicTuple {
-                    name: String::default(),
+                let info = self.get_represented_type_info();
+                DynamicTuple {
+                    name: Cow::Borrowed(::core::any::type_name::<Self>()),
```

Comment:
I think this is fine since we rely on `type_name` within `TypeInfo` anyways. And this is on the tuple implementation, meaning it won't have any custom name.